### PR TITLE
Add isPresent before completing consistency check

### PR DESCRIPTION
**Pull request by Manno15**
This will fix the issues found here: https://github.com/apache/accumulo/pull/2397#issuecomment-1046326584

The issue happened due to the optional(s)  still being empty while `lastFlushID` and `lastCompactID` were set to '0'.  This is not a case I ran into on an actual cluster so it might only be a fix for the specific IT that was failing. I did run some brief testing with this change and ran into no further issues.

### CODE REVIEW

**Comment by Manno15** [comment]
@ctubbsii This PR still has the commits from @foster33 before the revert. I am not quite sure how you would want to reapply that commit so I can create a new PR with just my specific changes if that is better.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java [hunk 1]
```diff
@@ -1401,6 +1401,22 @@ private void closeConsistencyCheck() {
...
+        throw new RuntimeException(msg);
+      }
+
+      if (tabletMeta.getCompactId().isPresent()
+          && tabletMeta.getCompactId().orElse(-1) != lastCompactID) {
+        String msg = "Closed tablet " + extent + " lastCompactID is inconsistent with metadata : "
+            + tabletMeta.getCompactId().orElse(-1) + " != " + lastCompactID;
+        log.error(msg);
+        throw new RuntimeException(msg);
+      }
```
[the remark this answers is not in the document]
Comment (Manno15) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java]: Your changes look good.

**Comment by ctubbsii** [comment]
> @ctubbsii This PR still has the commits from @foster33 before the revert. I am not quite sure how you would want to reapply that commit so I can create a new PR with just my specific changes if that is better.

Don't worry about it. Once I confirm all the ITs pass (and so far, it looks like they are headed that way), I'll just merge this PR in myself, squashed as usual, with references to the previous revert commit and issues, and with all the appropriate co-authored-by lines as well.

**Review by ctubbsii — APPROVED** [review]
All ITs passed, including the broken one. I'll merge it in now.

**Pull request merged**